ivadomed/losses.py

```diff
@@ -1,16 +1,59 @@
 import torch
 import torch.nn as nn
-import torch.nn.functional as F
 
 
-def dice_loss(input, target):
+# Inspired from https://arxiv.org/pdf/1802.10508.pdf
+def multi_class_loss(prediction, target, classes_of_interest=None):
+    n_classes = prediction.shape[1]
+    if classes_of_interest is None:
+        classes_of_interest = range(n_classes)
+    elif not len(classes_of_interest):
+        return 0.0
+
+    dice_per_class = 0
+    n = 0
+
+    for i in classes_of_interest:
+        dice_per_class += dice_function(prediction[:, i, ], target[:, i, ])
+        n += 1
+
+    return dice_per_class / n
+
+
+def dice_loss(prediction, target, params=None):
+    """
+    :param prediction: torch tensor generated as the model's prediction
+    :param target: torch tensor representing the ground truth
+    :param params: list of the losses considered for this loss
+    first item in list: represents classes of interest, list containing the index of a class which dice will be added to
+                        the loss.
+    second item in list: represents multi_class_dice, boolean representing whether or not multiclass dice is included in
+                         the current loss.
+    third item in list: represents dice, boolean representing whether or not dice is included in the current loss.
+
+    e.i: [[0, 1], True, False]
```

Comment:
If we send `[0, 1]` --> will it compute the dice of the background class?

Reply:
Oooo --> I have just realised [0, 1] refers to the two first classes as you remove the background after the softmax, correct?